pygfx/renderers/wgpu/_renderer.py

```diff
@@ -797,6 +797,7 @@ def convert_event(self, event: dict):
     "wheel": WheelEvent,
     "key_down": KeyboardEvent,
     "key_up": KeyboardEvent,
+    "before_draw": WindowEvent,
```

Comment:
I have a question about this; before_draw is a risky event to rely on, especially in applications without a steady frame rate. I assume you're aware of this and taking that into account?

Reply:
>  I assume you're aware of this and taking that into account?

Yes. The controller requests new draws when it know that it needs them (also when the mouse has been released, but the "animation" has not finished yet due to the smoothing). I'll make sure we have at least one example without external draw requests.

I will also look into before_draw in relation to viewports. I might create a separate PR for that.